metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_sequence_diagram.puml

```diff
@@ -0,0 +1,73 @@
+@startuml ci_integrations_config_validator_sequence_diagram
+
+skinparam ParticipantPadding 20
+skinparam BoxPadding 10
+
+actor Bob
+
+participant "CI Integrations"
+participant CiIntegrationsRunner
+participant SyncCommand
+participant ConfigValidatorFactory
+participant ConfigValidator
+participant ValidationDelegate
+participant CoolIntegrationClient
+
+Bob -> "CI Integrations" : runs sync
+activate "CI Integrations"
+
+"CI Integrations" -> CiIntegrationsRunner: run(List<String> arguments)
+activate CiIntegrationsRunner
+
+CiIntegrationsRunner -> SyncCommand: run()
+deactivate CiIntegrationsRunner
+activate SyncCommand
+
+SyncCommand -> SyncCommand : creates source and destination parties
+SyncCommand -> SyncCommand : parses source and destination configs
+
+SyncCommand -> ConfigValidatorFactory : create()
+activate ConfigValidatorFactory
+
+ConfigValidatorFactory --> SyncCommand : configValidator
+deactivate ConfigValidatorFactory
+
+SyncCommand -> ConfigValidator : validate(config)
+activate ConfigValidator
+
+
+ConfigValidator -> ValidationDelegate : tries to authenticate with the given credentials
+activate ValidationDelegate
+
+ValidationDelegate -> CoolIntegrationClient : authenticate()
+activate  CoolIntegrationClient
+CoolIntegrationClient --> ValidationDelegate : InteractionResult
+
+ValidationDelegate --> ConfigValidator : authentication result
+ConfigValidator -> ValidationDelegate : tries to validate other config fields
```

Comment:
Would it be great to display the case when the validation delegate returns an error interaction response here?